### DISCUSSION TITLE
Please remove dot from the air quality properties (PM2.5 -> PM2_5)

### DISCUSSION THREAD

**Discussion by koczkaandre** [arendst/Tasmota#14252] [Ideas]
JSON properties cannot be used in homekit2mqtt because of the dot in the json property. 
Removing the dot would enable the use of the json dot-notation (e.g airQuality.PM2_5). This would simplify code and troubleshooting for inexperienced users.

**Comment by sfromis** [comment]
One way of doing this, while retaining much of the backward compatibility, would be to make it dependent on the [SetOption64](https://tasmota.github.io/docs/Commands/#setoption64) value. While this is for replacing hyphen in sensor names by underscore, I think that the platforms having trouble with hyphen would often also have trouble with other characters not supposed to be in variable names, like period.

**Reply by koczkaandre** [reply]
Yes that's understandable. Then maybe "PM25" would be a solution.

**Reply by sfromis** [reply]
My point was that we *already* have an option to use underscore instead of a character making trouble for some types of backend software. Using the same to *optionally* replace the current period by underscore (`PM2_5`) could make sense, and this type of change would not be breaking to people using the key "PM2.5" without experiencing this as a problem.

Always removing the dot, or always replacing by underscore could be "as bad" to current users. My expectation is that people being fine with sensor names with a hyphen would typically also be fine with it containing a period. And people having trouble with one would likely also have trouble with the other, if they use affected sensor(s).
